betterconf/config.py

```diff
@@ -69,7 +59,16 @@ def value(self):
                 return self._default()
             else:
                 return self._default
-        return self._caster.cast(self._value)
+        casted = self._caster.cast(self._value)
+        if isinstance(casted, ImpossibleToCastError):
+            if self._ignore_caster_error:
+                return casted.val
+
+            else:
+                raise casted
```

Comment:
I see why you `return` the error now. But why not catch it and re-raise if `not self._ignore_caster_error` like this:
```python
try:
    return self._caster.cast(self._value)
except ImpossibleToCastError as e:
    if self._ignore_cast_error:
        return e.val
    else:
        raise e
```

Reply:
You are right. This is a better way